### PR TITLE
Fix rubocop offenses

### PR DESCRIPTION
**Pull request by bfcoder**
Update rubocop, remove erroneous cops, ignore specific files
Fix rubocop offenses

Most offenses were fixed with the autofix option (`rubocop -a`)

### CODE REVIEW

File: app/controllers/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -1,113 +1,131 @@
...
+      @user = current_user
+      auth = request.env["omniauth.auth"]
+      kind = params[:action].titleize # Should give us Facebook, Twitter, Linked In, etc
+      authentication = current_user.associate_account(auth)
+      current_user.save!
+      flash[:notice] = "Your #{Rails.application.secrets.application_name} account has been associated with #{kind}"
+      redirect_to after_sign_in_path_for(current_user) if should_redirect?
     end
+  rescue ActiveRecord::RecordInvalid => ex
+    message = error_message(ex, auth, authentication)
```
Comment (bfcoder) [inline app/controllers/omniauth_callbacks_controller.rb]: I extracted the code that was here to `#error_message`

File: spec/factories/users.rb [hunk 2]
```diff
@@ -1,20 +1,20 @@
 FactoryGirl.define do
-
   factory :user do
     name { FactoryGirl.generate(:name) }
     email { FactoryGirl.generate(:email) }
     password { FactoryGirl.generate(:password) }
-    after(:build) { |user| user.confirm }
+    after(:build, &:confirm)
```
Comment (bfcoder) [inline spec/factories/users.rb]: nice 👍

File: app/models/user.rb [hunk 3]
```diff
@@ -82,36 +101,36 @@ def set_default_role
     self.role ||= :user
   end
 
-  def is_in_role?(object, roles)
-    raise 'not implemented'
+  def in_role?(_object, _roles)
```
Comment (bfcoder) [inline app/models/user.rb]: I removed `is_` and no need to update elsewhere as it is not actually used

File: app/models/user.rb [hunk 4]
```diff
@@ -5,73 +5,92 @@ class User < ActiveRecord::Base
...
-      self.authentications.build(attributes)
+      authentications.build(attributes)
     end
   end
 
   def associate_account(auth)
-    data = auth['info'] || {}
-    name = data['name'] rescue nil
+    data = auth["info"] || {}
+    name = begin
```
Comment (jbasdf) [inline app/models/user.rb]: We can get rid of this block. data['name'] won't throw an exception.

File: app/models/user.rb (hunk 3 above)
Comment (jbasdf) [inline app/models/user.rb]: This method can be removed.

**Pull request merged**